cli/lib/exec/spawn.js

```diff
@@ -133,18 +148,55 @@ module.exports = {
       })
     }
 
-    const userFriendlySpawn = () => {
+    const spawnInXvfb = () => {
+      return xvfb
+      .start()
+      .then(() => {
+        // call userFriendlySpawn ourselves
+        // to prevent result of previous promise
+        // from becoming a parameter to userFriendlySpawn
+        return userFriendlySpawn()
+      })
+      .finally(xvfb.stop)
+    }
+
+    const userFriendlySpawn = (shouldRetryOnDisplayProblem) => {
+      debug('spawning, should retry on display problem?', Boolean(shouldRetryOnDisplayProblem))
+      if (os.platform() === 'linux') {
+        debug('DISPLAY is %s', process.env.DISPLAY)
+      }
+
       return spawn()
+      .then((code) => {
+        const POTENTIAL_DISPLAY_PROBLEM_EXIT_CODE = 234
+
+        if (shouldRetryOnDisplayProblem && code === POTENTIAL_DISPLAY_PROBLEM_EXIT_CODE) {
+          debug('Cypress thinks there is a display problem')
+          debug('retrying the command with our XVFB')
+
+          // if we get this error, we are on Linux and DISPLAY is set
+          logger.warn(`${stripIndent`
+
+            ${logSymbols.warning} Warning: Cypress process has finished very quickly with an error,
+            which might be related to a potential problem with how the DISPLAY is configured.
+
+            DISPLAY was set to "${process.env.DISPLAY}"
+
+            We will attempt to spin our XVFB server and run Cypress again.
+          `}\n`)
+
+          return spawnInXvfb()
+        }
+
+        return code
+      })
       .catch(throwFormErrorText(errors.unexpected))
     }
 
     if (needsXvfb) {
-      return xvfb.start()
-      .then(userFriendlySpawn)
-      .finally(xvfb.stop)
+      return spawnInXvfb()
     }
 
-    return userFriendlySpawn()
-
+    return userFriendlySpawn(os.platform() === 'linux')
```

Comment:
if we are on Linux, and there is a display problem, we should attempt handling it once